Code/Framework/AzCore/AzCore/Component/Component.cpp

```diff
@@ -163,6 +163,10 @@ namespace AZ
         }
     }
 
+    void Component::PrepareComponentForEntityAddition(Entity*)
+    {
+    }
```

Comment:
What was the issue with using SetEntity?

Reply:
2 issues primarily:
1. It's being used both when adding a component and removing a component. And even for adding, it's being called twice from EditorEntityActionComponent. So it was getting harder to test that the function only sets the alias once correctly
2. Since it's a public function, there are many calls to this already in the code base https://sourcegraph.com/search?q=context:global+repo:%5Egithub%5C.com/o3de/o3de%24+SetEntity&patternType=standard&sm=1&groupBy=path . We don't want components to call this manually to set an alias. So it felt like we are trying to piggyback our required behavior (to set a valid alias) on to an existing call that seems to have many use cases for many components.

Reply:
Thanks for the explanation. I understand the purpose of this function much better now.

How about given it a name that is more inline how we name similar functions in the code base.
Such as `OnBeforeEntityAdded` or `PreAddEntity`?

Reply:
That is fine.